utils/id.go

```diff
@@ -66,17 +73,91 @@ func LocalNodeID() (string, error) {
 	return fmt.Sprintf("%s%s", NodePrefix, HashedID(hostname)[:8]), nil
 }
 
-var b62Index = newB62Index()
-var b62Chars = []byte(shortuuid.DefaultAlphabet)
+var b57Index = newB57Index()
+var b57Chars = []byte(shortuuid.DefaultAlphabet)
 
-func newB62Index() [256]byte {
+func newB57Index() [256]byte {
 	var index [256]byte
-	for i := 0; i < len(b62Chars); i++ {
-		index[b62Chars[i]] = byte(i)
+	for i := 0; i < len(b57Chars); i++ {
+		index[b57Chars[i]] = byte(i)
 	}
 	return index
 }
 
+type guidGenerator struct {
+	pool sync.Pool
+	mu   sync.Mutex
+	buf  []byte
+	pos  int
+}
+
+func newGuidGenerator(bufSize, scratchSize int) *guidGenerator {
+	return &guidGenerator{
+		pool: sync.Pool{
+			New: func() any {
+				b := make([]byte, scratchSize)
```

Comment:
is there a need for `scratchSize` here? Trying to understand how this size works as a desired size slice is appended when used.

Reply:
i defined it as `GuidSize+10`. i think the biggest prefix might be `TR_RDC` but i didn't think we would miss the extra 4 bytes. if we need more `NewGuid` will grow the slice it's just less efficient.